lib/galaxy/tools/__init__.py

```diff
@@ -2252,16 +2252,7 @@ def populate_model(self, request_context, inputs, state_inputs, group_inputs, ot
 
     def _get_job_remap(self, job):
         if job:
-            if job.state == job.states.ERROR:
-                try:
-                    if [hda.dependent_jobs for hda in [jtod.dataset for jtod in job.output_datasets] if hda.dependent_jobs]:
-                        return True
-                    elif job.output_dataset_collection_instances:
-                        # We'll want to replace this item
-                        return 'job_produced_collection_elements'
-                except Exception as exception:
-                    log.error(str(exception))
-        return False
+            return job.remappable()
```

Comment:
I'd drop the method and replace its only call with `job.remappable() if job else None`